orchestra/proc-macro/src/impl_channels_out.rs

```diff
@@ -120,17 +135,27 @@ pub(crate) fn impl_channels_out_struct(info: &OrchestraInfo) -> Result<proc_macr
 				&mut self,
 				signals_received: usize,
 				message: #message_wrapper,
+				priority: ChannelsOutPriority,
 			) -> ::std::result::Result<(), #support_crate ::metered::TrySendError<#message_wrapper>> {
 				let res: ::std::result::Result<_, _> = match message {
 				#(
 					#feature_gates
 					#message_wrapper :: #consumes_variant ( inner ) => {
-						self. #channel_name .try_send(
-							#support_crate ::make_packet(signals_received, #maybe_boxed_send)
-						).map_err(|err| match err {
-								#support_crate ::metered::TrySendError::Full(err_inner) => #support_crate ::metered::TrySendError::Full(#message_wrapper:: #consumes_variant ( #maybe_unbox_error )),
-								#support_crate ::metered::TrySendError::Closed(err_inner) => #support_crate ::metered::TrySendError::Closed(#message_wrapper:: #consumes_variant ( #maybe_unbox_error )),
-						})
+						if matches!(priority, ChannelsOutPriority::High) {
```

Comment:
In case https://github.com/paritytech/orchestra/pull/77/files#r1627665325 is not implemented, please create a full `match` statement to not accidentally break code like this in the future. Most of the code appears to be duplicated, i.e. `.map_err` doesn't have to life within the branches, but can be used on the return value of the expression, `try_priority_send` and `try_send` return the same `Result<T,E>` type iiuc.